src/main/kotlin/com/hibob/kotlinBasics/types/StoreService.kt

```diff
@@ -0,0 +1,44 @@
+package com.hibob.academy.types
+
+import com.hibob.kotlinBasics.types.*
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+       return cart.map({it -> it.clientId to checkout(it, payment)}).toMap()
+
+    }
+
+    fun checkout(cart: Cart, payment: Payment):Check {
+        val total = cart.products.filter { isCustomTrue(it) }.sumOf { it.price }
+
+        if(!validatePaymentMethod(payment, total)){
+            return Check(cart.clientId, Statuses.FAILURE, 0.0)
+        }
+        return Check(cart.clientId, Statuses.SUCCESS, total)
+    }
+
+    fun validatePaymentMethod(payment: Payment, total: Double): Boolean {
+        return when (payment) {
+            is Payment.Cash -> fail("invalid payment method")
+            is Payment.PayPal -> payment.email.contains("@")
+            is Payment.CreditCard -> validateCard(payment, total)
+        }
+    }
+
+    fun validateCard(card : Payment.CreditCard, total: Double): Boolean {
+        return (card.type == CreditCardType.VISA || card.type == CreditCardType.MASTERCARD) && card.limit > total &&
+                card.number.length > 9 && card.expiryDate.isAfter(LocalDate.now())
```

Comment:
first you need to veirfy length == 10
second this condition is not clear first I suggest you in case of CreditCard to call function called validateCreditCard.
then I would suggest you to split the condition to parts, and each part to use function that going to be more readable and easier to fix.

something like:
  
```
private fun validateCreditCard(card: Payment.CreditCard, total: Double) =
        isSupportedType(card.type) && isCardDetailsValid(card) && isNotOverTheLimit(price, card)

    private fun isSupportedType(cardType: CreditCardType) =
        (cardType == CreditCardType.MASTERCARD || cardType == CreditCardType.VISA)

    private fun isCardDetailsValid(payment: Payment.CreditCard) =
        payment.expiryDate.isAfter(LocalDate.now())  && payment.number.length == 10

    private fun isNotOverTheLimit(price: Double, payment: Payment.CreditCard) = price < payment.limit
```
look at the condition now, I think its easier to read it